Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -554,7 +562,23 @@ void SourceBufferPrivate::didReceiveInitializationSegment(SourceBufferPrivateCli
         return;
     }
 
-    m_client->sourceBufferPrivateDidReceiveInitializationSegment(WTFMove(segment), WTFMove(completionHandler));
+    m_processingInitializationSegment = true;
+    m_client->sourceBufferPrivateDidReceiveInitializationSegment(WTFMove(segment), [this, weakThis = WeakPtr { *this }, completionHandler = WTFMove(completionHandler)] (auto result) mutable {
+        if (!weakThis) {
+            completionHandler(result);
+            return;
+        }
+        m_processingInitializationSegment = false;
+        processPendingSamples();
```

Comment:
You always clear `m_processingInitializationSegment` before calling `processPendingSamples`, maybe do it in `processPendingSamples` instead?

Reply:
Is there ever a case where `m_processingInitializationSegment` is true in `processPendingSamples()`